### ISSUE TITLE
NetworkXError: G is not a connected graph. metric_closure is not defined.

### ISSUE THREAD

**Issue by daren-thomas** [architecture-building-systems/CityEnergyAnalyst#1823]
When trying to run the `cea network-layout` tool using the cooling case study (e.g. `C:\reference-case-WTP\WTP_CBD_h`, I get the following error:

```
Traceback (most recent call last):
  File "C:\Users\darthoma\Miniconda2\envs\cea\lib\multiprocessing\process.py", line 267, in _bootstrap
    self.run()
  File "C:\Users\darthoma\Miniconda2\envs\cea\lib\multiprocessing\process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\interfaces\dashboard\tools\worker.py", line 37, in run_script
    script(**kwargs)
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\api.py", line 26, in script_runner
    script_module.main(config)
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\technologies\thermal_network\network_layout\main.py", line 57, in main
    network_layout(config, locator, plant_building_names)
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\technologies\thermal_network\network_layout\main.py", line 50, in network_layout
    optimization_flag, plant_building_names, disconnected_building_names)
  File "c:\users\darthoma\documents\github\cityenergyanalyst\cea\technologies\thermal_network\network_layout\steiner_spanning_tree.py", line 66, in calc_steiner_spanning_tree
    mst_non_directed = nx.Graph(steiner_tree(G, terminal_nodes))
  File "", line 2, in steiner_tree
  File "C:\Users\darthoma\Miniconda2\envs\cea\lib\site-packages\networkx\utils\decorators.py", line 82, in _not_implemented_for
    return not_implement_for_func(*args, **kwargs)
  File "", line 2, in steiner_tree
  File "C:\Users\darthoma\Miniconda2\envs\cea\lib\site-packages\networkx\utils\decorators.py", line 82, in _not_implemented_for
    return not_implement_for_func(*args, **kwargs)
  File "C:\Users\darthoma\Miniconda2\envs\cea\lib\site-packages\networkx\algorithms\approximation\steinertree.py", line 82, in steiner_tree
    M = metric_closure(G, weight=weight)
  File "", line 2, in metric_closure
  File "C:\Users\darthoma\Miniconda2\envs\cea\lib\site-packages\networkx\utils\decorators.py", line 82, in _not_implemented_for
    return not_implement_for_func(*args, **kwargs)
  File "C:\Users\darthoma\Miniconda2\envs\cea\lib\site-packages\networkx\algorithms\approximation\steinertree.py", line 35, in metric_closure
    raise nx.NetworkXError(msg)
NetworkXError: G is not a connected graph. metric_closure is not defined.
```

@JIMENOFONSECA do you have an idea what could be the problem?

**Comment by JIMENOFONSECA** [comment]
Yes, the problem is being solved in the fix-optimization branch
On 13 Mar 2019, 20:23 +0800, Daren Thomas <notifications@github.com>, wrote:

Assigned #1823<https://github.com/architecture-building-systems/CityEnergyAnalyst/issues/1823> to @JIMENOFONSECA<https://github.com/JIMENOFONSECA>.

**Comment by JIMENOFONSECA** [comment]
fixed in #1810
